src/shared/telemetry/telemetryUtils.ts

```diff
@@ -9,6 +9,19 @@ import * as vscode from 'vscode'
 import { ext } from '../extensionGlobals'
 import { Datum } from './telemetryEvent'
 
+export interface TelemetryName {
+    namespace: TelemetryNamespace
+    name: string
+}
+
+export enum TelemetryNamespace {
```

Comment:
nit: It seems weird (to me) to export this as namespace, but I can't think of a more apt name either

Reply:
I do like that this is finally an enum though!

Reply:
I was torn between `TelemetryNamespace` and `TelemetryNamespaces`; the convention through the rest of the codebase is singular for `enum`s though.